infrastructure/terraform/per_workspace/modules/etl/sds/etl-diagram--update-transform-and-load.asl.json

```diff
@@ -1,91 +1,113 @@
+                  "Lambda.TooManyRequestsException"
+                ],
+                "IntervalSeconds": 1,
+                "MaxAttempts": 3,
+                "BackoffRate": 2
               }
-            }
-          ],
-          "Next": "load"
-        }
-      ],
-      "Default": "transform"
-    },
-    "transform": {
-      "Type": "Task",
-      "Resource": "arn:aws:states:::lambda:invoke",
-      "OutputPath": "$.Payload",
-      "Parameters": {
-        "FunctionName": "${transform_worker_arn}:$LATEST",
-        "Payload": {
-          "max_records": "1",
-          "trust": false
-        }
-      },
-      "Retry": [
-        {
-          "ErrorEquals": [
-            "Lambda.ServiceException",
-            "Lambda.AWSLambdaException",
-            "Lambda.SdkClientException",
-            "Lambda.TooManyRequestsException"
-          ],
-          "IntervalSeconds": 1,
-          "MaxAttempts": 3,
-          "BackoffRate": 2
+            ],
+            "Next": "load-result"
+          },
+          "load-result": {
+            "Type": "Choice",
+            "Choices": [
+              {
+                "Not": {
+                  "Variable": "$.error_message",
+                  "IsNull": true
+                },
+                "Next": "etl-stopped"
+              },
+              {
+                "And": [
+                  {
+                    "Variable": "$.error_message",
+                    "IsNull": true
+                  },
+                  {
+                    "Not": {
+                      "Variable": "$.unprocessed_records",
+                      "NumericEquals": 0
+                    }
+                  }
+                ],
+                "Next": "load"
+              }
+            ],
+            "Default": "transform"
+          },
+          "transform": {
+            "Type": "Task",
+            "Resource": "arn:aws:states:::lambda:invoke",
+            "OutputPath": "$.Payload",
+            "Parameters": {
+              "FunctionName": "${transform_worker_arn}:$LATEST",
+              "Payload": {
+                "max_records": "1",
+                "trust": false
+              }
+            },
+            "Retry": [
+              {
+                "ErrorEquals": [
+                  "Lambda.ServiceException",
+                  "Lambda.AWSLambdaException",
+                  "Lambda.SdkClientException",
+                  "Lambda.TooManyRequestsException"
+                ],
+                "IntervalSeconds": 1,
+                "MaxAttempts": 3,
+                "BackoffRate": 2
+              }
+            ],
+            "Next": "etl-stopped"
+          },
+          "etl-stopped": {
+            "Type": "Pass",
+            "End": true
+          }
         }
-      ],
-      "Next": "etl-stopped"
+      }
     },
-
-    "etl-stopped": {
+    "pass": {
       "Type": "Pass",
-      "End": true
+      "End": true,
+      "OutputPath": "$[9]"
```

Comment:
What happens if it does't need to loop 10 times? Or will it always loop the full amount and split the changes/events evenly?

Reply:
it will always loop the full amount, and if any are not needed then they effectively won't do anything